contracts/seed/Seed.sol

```diff
@@ -361,16 +343,18 @@ contract Seed {
       * @dev                     Remove address from whitelist.
       * @param buyer             Address which needs to be unwhitelisted
     */
-    function unwhitelist(address buyer) public onlyAdmin isActive {
-        require(permissionedSeed == true, "Seed: module is not whitelisted");
+    function unwhitelist(address buyer) external onlyAdmin {
+        require(!closed, "Seed: should not be closed");
+        require(permissionedSeed == true, "Seed: seed is not whitelisted");
 
         whitelisted[buyer] = false;
     }
 
     /**
       * @dev                     Withdraw funds from the contract
     */
-    function withdraw() public onlyAdmin allowedToWithdraw {
+    function withdraw() external onlyAdmin {
+        require(minimumReached, "Seed: minimum funding amount not met");
```

Comment:
admin should be able to withdraw at any point after buying is no longer possible

Reply:
you mean, admin cannot withdraw funding tokens when the buy() is active, even after minimumReached==true i.e. when funding tokens cannot be claimed by the funders/users/buyers.

Reply:
if so, then the condition when withdraw is allowed would be
when `block.timestamp > vestingStartTime`
@dkent600

Reply:
same as above:

 require(closed || maxReached || (minReached && (blocktime >= endTime)))

Reply:
done

Reply:
we cannot allow withdraw when closed == true,
Scenario:-
1. A user buys seed token.
2. admin closes the Seed contract, but the minimum target is not reached.
3. admin can still withdraw the funding tokens, which should not be allowed.

Admin can withdraw under the following conditions,
1. Maximum target reached
2. Minimum target reached and `blockTime >= endTime`